### PR TITLE
store magnetic field in output

### PR DESCRIPTION
**Pull request by selvaggi**
BEGINRELEASENOTES
- store magnetic field in output

ENDRELEASENOTES

### CODE REVIEW

File: converter/src/DelphesEDM4HepConverter.cc [hunk 1]
```diff
@@ -189,6 +190,10 @@ namespace k4SimDelphes {
     //UserData for overflowing information
     auto* pathLengthCollection = createCollection<podio::UserDataCollection<float>>(branch + "_L");
 
+    // user data to define magnetic field (fill once per event)
+    auto* magFieldCollection = createCollection<podio::UserDataCollection<float>>("B");
```
Comment (tmadlener) [inline converter/src/DelphesEDM4HepConverter.cc:194]: Should we give this a slightly more descriptive name? I would at least add a `z` to indicate the direction in this case.

**Review by tmadlener — COMMENTED** [review]
Very nice addition. Technically we could also store this as a parameter in the Frame (at least for standalone). However, that would require some more work when accessing it in RDataFrame / FCCanalyses potentially.

File: converter/src/DelphesEDM4HepConverter.cc (hunk 1 above)
Comment (selvaggi) [reply converter/src/DelphesEDM4HepConverter.cc]: indeed, thanks for the comment, done!

**Comment by andresailer** [comment]
Somehow I feel people should not get used to finding this kind of information in the (meta)data, but from the geometry service or via some utility function. Otherwise analyses that want to compare the delphes and full simulation output would need some kind of switch.

File: converter/src/DelphesEDM4HepConverter.cc [hunk 2]
```diff
@@ -10,6 +10,7 @@
 #include "edm4hep/ReconstructedParticleCollection.h"
 #include "edm4hep/TrackCollection.h"
 #include "edm4hep/TrackerHitCollection.h"
+#include "edm4hep/CalorimeterHitCollection.h"
```
Comment (andresailer) [inline converter/src/DelphesEDM4HepConverter.cc:13]: This is already included above.

Comment (selvaggi) [reply converter/src/DelphesEDM4HepConverter.cc]: fixed.

Comment (selvaggi) [reply converter/src/DelphesEDM4HepConverter.cc]: Hello @andresailer  @tmadlener anything that prevents this to merged?

**Pull request merged**